modules/core/src/main/java/org/apache/ignite/hlc/HybridClock.java

```diff
@@ -40,13 +47,25 @@ public class HybridClock {
     }
 
     /** Latest timestamp. */
-    private volatile HybridTimestamp latestTime;
+    protected volatile HybridTimestamp latestTime;
 
     /**
      * The constructor which initializes the latest time to current time by system clock.
      */
     public HybridClock() {
-        this.latestTime = new HybridTimestamp(Clock.systemUTC().instant().toEpochMilli(), 0);
+        this.currentTimeMillisProvider = () -> Clock.systemUTC().instant().toEpochMilli();
+
+        this.latestTime = new HybridTimestamp(currentTimeMillisProvider.get(), 0);
+    }
+
+    /**
+     * The constructor which initializes the latest time to the given provider.
+     */
+    @TestOnly
```

Comment:
I don't like an idea of adding test related code to HybridClock, neither @TestOnly nor protected along with all suppliers.